src/Randomizer.SMZ3/Regions/Zelda/TurtleRock.cs

```diff
@@ -92,13 +92,8 @@ public TurtleRock(World world, Config config) : base(world, config)
 
         public override bool CanEnter(Progression items)
         {
-            return Medallion switch
-            {
-                ItemType.Bombos => items.Bombos,
-                ItemType.Ether => items.Ether,
-                _ => items.Quake
-            } && items.Sword &&
-                items.MoonPearl && Logic.CanLiftHeavy(items) && items.Hammer && items.Somaria &&
+            return ((Medallion == ItemType.Bombos && items.Bombos) || (Medallion == ItemType.Ether && items.Ether) || (Medallion == ItemType.Quake && items.Quake)) &&
```

Comment:
Inlining the switch expression in a boolean condition like is kind of weird, but this new condition isn't much better. Considering this is the second spot now, I'm in favor of putting this particular check in the `Progression` class or something, like a `HasMedallion(items, medallion)`.

That would keep this whole thing readable and would let you use a regular switch case statement if the new switch expression isn't supported.

Reply:
So actually, since the Medallion is just an ItemType, I was able to use items.Contains(Medallion).